donkeycar/parts/imu.py

```diff
@@ -30,7 +57,14 @@ def update(self):
                 
     def poll(self):
         try:
-            self.accel, self.gyro, self.temp = self.sensor.get_all_data()
+            if self.sensortype == SENSOR_MPU6050:
+                self.accel, self.gyro, self.temp = self.sensor.get_all_data()
+            else:
+                from mpu9250_jmdev.registers import GRAVITY
+                ret = self.sensor.getAllData()
+                self.accel = { 'x' : ret[1] * GRAVITY, 'y' : ret[2] * GRAVITY, 'z' : ret[3] * GRAVITY }
+                self.gyro = { 'x' : ret[4], 'y' : ret[5], 'z' : ret[6] }
+                self.mag = { 'x' : ret[13], 'y' : ret[14], 'z' : ret[15] }
```

Comment:
You add magnetometer  data in here but it is unused at all, because in ` donkeycar/templates/complete.py` in line 268 you only use inputs from accelerometer and gyroscope.

Also AFAIR if adding magnetometer data is ok, but in reality it may be confusing for trained model if someone moves to another location (like from EU to USA).
So I think that for driving car the magnetometer data can be dropped (as it is now).

Otherwise it could be exposed if someone would like to use this data for something like geospatial vizualization?

Reply:
I'm going to progress in several steps about the magnetometer. I have previously worked with drones using several MPU family members. For this reason I had to deal a lot with  magnetometers. 
For this reason the first step is collecting data e trying to understand the quality of what i can get from it. I want to understand the effect of surrounding elements as well (so entity of both soft/hard iron distortion without complex imu installations). this will help me understand how usable it is and if I actually need a proper calibration to get something useful from it.

Reply:
Keep the `magnetometer` data. Even though it's unused for know - the sensor data should be available for future models & approaches.